### ISSUE TITLE
Instanced type "virtual machine" failed not supported on this server

### ISSUE THREAD

**Issue by nixfreak** [beringresearch/macpine#51]
I was able to get containers running no problem with your setup but I am looking to see if I can replace Virtual Box, but I need to be able to create actual VM's to do that. 

Is Alpine supported to create VM's  through LXC ?

**Comment by idroz** [comment]
I think it should be possible. See set up here: https://beringresearch.github.io/macpine/lxd_macpine/

The above links to creation of LXD containers, but I think it should also be possible to launch LXD VMs.

**Comment by idroz** [comment]
I think this will require nested virtualisation through QEMU by tinkering with its `-cpu` parameters. Will need to investigate.

**Comment by maxzinkus** [comment]
This may be out of scope for this project. `macpine` seems to be dedicated to creating VMs from a single base Alpine Linux image rather than to replace a general-purpose VM manager. The [nemu project](https://github.com/nemuTUI/nemu) may be more appropriate for console VM management using `qemu`.

**Comment by maxzinkus** [comment]
@idroz thoughts on this? This issue seems like a large undertaking and maybe it's best to close it out as wontfix so people don't wait on it.

**Comment by maxzinkus** [comment]
And @nixfreak if you are around and have thoughts on this, please let us know!

**Comment by idroz** [comment]
There's a degree of case-specific customisation, which will be required.

1. Running nested VMs https://graspingtech.com/nested-virtualization-mac/
2. Running VMs inside containers https://discuss.linuxcontainers.org/t/lxc-vm-inside-lxc-container/14254/3

My view is that at this stage this is out of scope as the idea is to be as minimal and general as possible.

The best alternative to replace virtual box on Mac OS is probably https://mac.getutm.app

**Comment by idroz** [comment]
I'm not sure nested VMs are in scope at this stage. Will close as resolved. **However**, will keep in mind as there are a few things within LXD VMs that appear to be making this processes easier (e.g. https://discuss.linuxcontainers.org/t/lxd-vm-nested-virtualization/14615).

Feel free to reopen if critical.
